include/xrpl/protocol/PublicKey.h

```diff
@@ -128,8 +129,28 @@ class PublicKey
     {
         return slice();
     }
+
+    Slice
+    shortId() const noexcept
+    {
+        return slice().substr(0, 10);
```

Comment:
I noticed that Node IDs are typically represented using base58 encoding, generated by `toBase58(TokenType::NodePublic, ...)`. What do you think about following that convention here to maintain consistency?